src/vcpkg/visualstudio.cpp

```diff
@@ -45,6 +90,44 @@ namespace vcpkg::VisualStudio
     static constexpr CStringView V_142 = "v142";
     static constexpr CStringView V_143 = "v143";
 
+    static constexpr CStringView WORKLOAD_NATIVE_DESKTOP = "Microsoft.VisualStudio.Workload.NativeDesktop";
+    static constexpr CStringView WORKLOAD_CORE_FEATURES_TOOLS =
+        "Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Core";
+    static constexpr CStringView WORKLOAD_MSBUILD = "Microsoft.Component.MSBuild";
+    static constexpr CStringView COMPONENT_VCTOOLS_2015 = "Microsoft.VisualStudio.Component.VC.140";
+    static constexpr CStringView COMPONENT_VCTOOLS_2017_OR_LATER = "Microsoft.VisualStudio.Component.VC.Tools.x86.x64";
+    static constexpr CStringView COMPONENT_UCRT = "Microsoft.VisualStudio.Component.Windows10SDK";
+    static constexpr CStringView COMPONENT_ARM_VCTOOLS = "Microsoft.VisualStudio.Component.VC.Tools.arm";
+    static constexpr CStringView COMPONENT_ARM64_VCTOOLS = "Microsoft.VisualStudio.Component.VC.Tools.arm64";
+    static constexpr CStringView COMPONENT_UWP = "Microsoft.VisualStudio.ComponentGroup.UWP.VC";
+
+    struct WINDOWS_SDK
+    {
+        CStringView sdk_component_name;
+        CStringView sdl_version_string;
+    };
+
+    static constexpr WINDOWS_SDK WINDOWS_SDKS[] = {{"Windows81SDK", "8.1"},
+                                                   {"Windows10SDK.18362", "18362"},
+                                                   {"Windows10SDK.19041", "19041"},
+                                                   {"Windows10SDK.20348", "20348"},
+                                                   {"Windows11SDK.22000", "22000"}};
```

Comment:
I don't think we should hardcode Windows SDK versions here. We should instead detect components starting with "Windows81SDK"/"Windows10SDK"/"Windows11SDK".

Reply:
See [my comment](https://github.com/microsoft/vcpkg-tool/pull/314#discussion_r785315261).